### DISCUSSION TITLE
Can use stackusage to show the running process stack status?

### DISCUSSION THREAD

**Discussion by xixiguo** [d99kris/stackusage#15] [Q&A]
If the process does not exit and running, how to see the stackusage output?
I tried to use the option -o outputfile. But I found that it just show the main process state as below,
**_stackusage log at 2022-03-12 04:30:37 ----------------------------------------
  pid       id        tid       requested            actual       maxuse      max%        dur               funcP name
  394       0       394         8388608        8384512           3856             0            0               (nil) i2cset_**
This result seems did not include the states of pthreads created by the main process. But the main process created several pthreads. Is there anyone can help?

**Comment by d99kris** [comment]
Hi @xixiguo - yes it should be possible. From the README:
```
stackusage -s SIGUSR1 gedit
kill -s SIGUSR1 `pidof gedit`
```
This also works if you used `-o` option, at least in my simple `gedit` test, like:
```
stackusage -o log.txt -s SIGUSR1 gedit
kill -s SIGUSR1 `pidof gedit`
```
So basically you need to use `kill` to signal stackusage to log the current state. Is that what you tried too? Perhaps you can share the exact commands you used to run `stackusage` as well as how you called `kill`, if it's still not working. I haven't tried it on `i2cset`, but I don't think there's anything special with `i2cset` - unless it has some special signal handling overriding/preventing use of SIGUSR1, SIGUSR2 or others.
